drone/src/agent/mod.rs

```diff
@@ -145,8 +164,14 @@ async fn listen_for_drain(
         tracing::info!(req=?req.message(), "Received request to drain drone.");
         req.respond(&()).await?;
 
-        send_ready
-            .send(!req.value.drain)
+        let state = if req.value.drain {
+            DroneState::Draining
+        } else {
+            DroneState::Ready
```

Comment:
I wonder if it could become an issue that sending a `drain: false` message sets a drone state to `Ready`? I suppose we could make sure we only set `DroneState::Ready` if `req.value.drain == false` _and_ the current state is `Draining`? Likewise, we'd only set `DroneState::Draining` if `req.value.drain == true` _and_ the current state is `Ready`. Is that too defensive? Thoughts?

Reply:
Good catch. I'm ok with this for now because setting drain to false is something we do manually, but I'll give it some thought.